docs/source/webhooks.rst

```diff
@@ -173,22 +173,62 @@ When Not to Use Webhooks
 
 While webhooks are useful, they do have two drawbacks:
 
-* **Not Bidirectional**  - Webhooks simply send data into stackstorm. So if you want data back from   StackStorm, or an action, you'll have to get that data in an asynchronous fashion
-* **No Guarantee of Execution** - Webhooks in StackStorm do not guarantee an execution. This all depends on configuration of the rule that registered the webhook.
+* **Not Bidirectional**  - Webhooks simply submit data into |st2|. So if you want data back from |st2|,
+  or an action execution ID, you'll have to get that data in an asynchronous fashion.
+* **No Guarantee of Execution** - Webhooks in StackStorm do not guarantee an execution. It depends on
+  the configuration of the rule that registered the webhook - based upon the webhook contents, it may
+  not execute any action, or may execute multiple actions.
 
-If you're looking for guaranteed responses, it might be better to directly leverage the
-execution API to directly run actions and query for a response, in the same way the ``st2 run``
-command does behind the scenes. We can get a little insight into this using the ``--debug`` flag:
+If you always want to execute a specific action or workflow, and/or you're looking for guaranteed response,
+you could leverage the ``/v1/executions`` API. This is the same as running an action from the CLI with
+``st2 run <mypack>.<myaction>``. We can get a little insight into this using the ``--debug`` flag:
 
 .. sourcecode:: bash
 
     st2 --debug run core.local "date"
+    2017-03-31 08:21:18,706  DEBUG - Using cached token from file "/home/ubuntu/.st2/token-st2admin"
```

Comment:
I was going to stay away from this because of the length, but after seeing how this is rendered, it's not so bad. If users really want to see this output they can scroll to the right. Or actually run the commands for themselves, which is visible in the text window.